src/preamble.js

```diff
@@ -1083,6 +1095,9 @@ function createWasm(env) {
       }
       return false;
     }
+#if LOAD_SOURCE_MAP
+    Module['sourceMap'] = new WASMSourceMap(getSourceMap());
```

Comment:
how can this work synchronously? I think we can just error in this case, as mentioned above.

Reply:
Why wouldn't it work synchronously? It should work synchronously on shells and perhaps node.js.